### PR TITLE
Add module `SAWScript.Panic` with bindings to `panic` library.

### PR DESCRIPTION
**Pull request by brianhuffman**
The new module is based on `Cryptol.Utils.Panic` from cryptol; it provides a function `panic :: String -> [String] -> a` that produces error messages like this:
```
You have encountered a bug in Saw's implementation.
*** Please create an issue at https://github.com/GaloisInc/saw-script/issues

%< ---------------------------------------------------
  Revision:  f7e4738b0648af6fe788bfa937fcd8610dd3940f
  Branch:    panic (uncommited files present)
  Location:  <user-provided string>
  Message:   <user-provided string>
CallStack (from HasCallStack):
  panic, called at src/SAWScript/Panic.hs:20:9 in saw-script-0.6.0.99-1VGCoxQ2nNV4ntdMbYXftm:SAWScript.Panic
  panic, called at src/SAWScript/Crucible/JVM/Builtins.hs:897:22 in saw-script-0.6.0.99-1VGCoxQ2nNV4ntdMbYXftm:SAWScript.Crucible.JVM.Builtins
%< ---------------------------------------------------
```

### CODE REVIEW

File: src/SAWScript/Panic.hs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+{-# LANGUAGE Trustworthy, TemplateHaskell #-}
+module SAWScript.Panic
+  (HasCallStack, SawPanic, Saw, Panic, panic) where
+
+import Panic hiding (panic)
+import qualified Panic as Panic
+
+data Saw = Saw
+
+type SawPanic = Panic Saw
```
Comment (andreistefanescu) [inline src/SAWScript/Panic.hs:17]: why is this needed?

**Review by andreistefanescu — APPROVED** [review]
👍

File: src/SAWScript/Panic.hs (hunk 1 above)
Comment (brianhuffman) [reply src/SAWScript/Panic.hs:17]: I'm not sure; I copied it from cryptol. Maybe we don't need it.

File: src/SAWScript/Panic.hs [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  HasCallStack =>
+  -- | Location of problem
+  String ->
+  -- | Problem description (lines)
+  [String] ->
+  a
+panic = Panic.panic Saw
+
+instance PanicComponent Saw where
+  panicComponentName _ = "Saw"
```
Comment (kquick) [inline src/SAWScript/Panic.hs:30]: This should probably be "SAW" since that's the user-visible name (and to avoid confusion).  I don't see it in your sample output, but I'm assuming this is displayed somewhere.

**Review by kquick — APPROVED** [review]

File: src/SAWScript/Panic.hs (hunk 2 above)
Comment (brianhuffman) [reply src/SAWScript/Panic.hs]: I was kind of thinking the same thing.
